core/util/runPuppet.js

```diff
@@ -118,6 +118,19 @@ async function processScenarioView (scenario, variantOrScenarioLabelSafe, scenar
     }
     await page.goto(translateUrl(url));
 
+		//  --- LOCAL STORAGE SCRIPT ---
```

Comment:
Please implement as an add in -- like `loadCookies` -- see: `capture/engine_scripts/puppet/onBefore.js`

Do not implement this on the main codepath.